pkg/module_manager/apis/v1alpha1/validate.go

```diff
@@ -0,0 +1,46 @@
+package v1alpha1
+
+import (
+	"context"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+	kwhhttp "github.com/slok/kubewebhook/v2/pkg/http"
+	kwhlogrus "github.com/slok/kubewebhook/v2/pkg/log/logrus"
+	"github.com/slok/kubewebhook/v2/pkg/model"
+	kwhvalidating "github.com/slok/kubewebhook/v2/pkg/webhook/validating"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen=false
+var vf = kwhvalidating.ValidatorFunc(func(ctx context.Context, review *model.AdmissionReview, obj metav1.Object) (result *kwhvalidating.ValidatorResult, err error) {
+	// UserInfo groups: [system:serviceaccounts system:serviceaccounts:d8-system system:authenticated]
+	// Extra: [authentication.kubernetes.io/pod-name:[deckhouse-7cf6ddd78-grn4m]]
+	if review.UserInfo.Username != "system:serviceaccount:d8-system:deckhouse" {
```

Comment:
I started thinking that maybe instead of introducing a validating webhook, it is better to raise an APIService, but this approach also has its pros and cons, e.g., not being able to control the apiVersion of Module objects.

Reply:
I fully agree but it seems a bit raw at the moment. I think we can discuss this idea later, in the another iteration